vpr/src/route/lookahead_profiler.h

```diff
@@ -0,0 +1,80 @@
+#ifndef VTR_LOOKAHEAD_PROFILER_H
+#define VTR_LOOKAHEAD_PROFILER_H
+
+#include <fstream>
+
+#include "connection_router_interface.h"
+#include "router_lookahead.h"
+#include "rr_graph_fwd.h"
+
+/**
+ * @brief A class which records information used to profile the router lookahead: most importantly,
+ * the actual cost (delay and congestion) from nodes to the sink to which they have been routed, as
+ * well as the lookahead's estimation of this cost.
+ *
+ * @warning
+ * To use the LookaheadProfiler, you must build VPR with #define PROFILE_LOOKAHEAD.
+ */
+class LookaheadProfiler {
+  public:
+    LookaheadProfiler()
+        : enabled_(false) {}
+
+    LookaheadProfiler(const LookaheadProfiler&) = delete;
+    LookaheadProfiler& operator=(const LookaheadProfiler&) = delete;
+
+    /**
+     * @brief Set the name of the output file.
+     *
+     * @note
+     * Passing in an empty string will disable the profiler.
+     *
+     * @param file_name The name of the output csv file.
+     */
+    void set_file_name(const std::string& file_name);
+
+    /**
+     * @brief Record information on nodes on a path from a source to a sink.
+     *
+     * @param iteration The router iteration.
+     * @param target_net_pin_index Target pin of this sink in the net.
+     * @param cost_params
```

Comment:
Suggest briefly commenting each input's purpose.

Reply:
You need a high-level comment of how the lookahead profiler works somewhere (file level, or maybe on this routine).